server/channels/app/post_persistent_notification.go

```diff
@@ -0,0 +1,367 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/mattermost/mattermost-server/v6/server/channels/app/request"
+	"github.com/mattermost/mattermost-server/v6/server/channels/store"
+	"github.com/mattermost/mattermost-server/v6/server/platform/shared/mlog"
+	"github.com/pkg/errors"
+)
+
+// DeletePersistentNotificationsPost stops persistent notifications, if loggedInUserID(except post owner) reacts, reply or ack on the post.
+// Post-owner can only delete the original post to stop the notifications, in which case "checkMentionedUser" must be "false".
+func (a *App) DeletePersistentNotificationsPost(c request.CTX, post *model.Post, loggedInUserID string, checkMentionedUser bool) *model.AppError {
+	if !a.IsPersistentNotificationsEnabled() {
+		c.Logger().Debug("DeletePersistentNotificationsPost: Persistent Notification feature is not enabled.")
```

Comment:
It's a debug level. But I'll remove it as it is in a "hot path".

Reply:
Even at a debug level, it's not consistently tied to a triggering event. If, for example, we only logged when trying to acknowledge a persistent notification, it would make sense to explain why we didn't do anything.